internal/util/util.go

```diff
@@ -60,15 +65,51 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	if opts.Type == "image" {
+		if opts.Attrs["push"] != "true" {
+			return nil
+		}
+
+		image := opts.Attrs["name"]
+		destSpec := opts.Attrs["name"]
+		dest, err := alltransports.ParseImageName(destSpec)
+		// add the docker:// transport to see if they neglected it.
+		if err != nil {
+			destTransport := strings.Split(destSpec, ":")[0]
+			if t := transports.Get(destTransport); t != nil {
+				return err
+			}
+
+			if strings.Contains(destSpec, "://") {
+				return err
+			}
+
+			destSpec = "docker://" + destSpec
+			dest2, err2 := alltransports.ParseImageName(destSpec)
```

Comment:
If this is all that is necessary, use `docker.ParseReference` without involving `alltransports`.

Reply:
https://github.com/containers/buildah/pull/4677#discussion_r1174183375